bin/bats

```diff
@@ -1,3 +1,35 @@
 #! /usr/bin/env bash
 
-exec "${0%/*}/../libexec/bats" "$@"
+set -e
+
+export BATS_READLINK='true'
+if command -v 'greadlink' >/dev/null; then
+  BATS_READLINK='greadlink'
+elif command -v 'readlink' >/dev/null; then
+  BATS_READLINK='readlink'
+fi
+
+bats_resolve_link() {
+  if ! "$BATS_READLINK" "$1"; then
+    return 0
+  fi
+}
+
+bats_absolute_path() {
+  local cwd="$PWD"
+  local path="$1"
+
+  while [[ -n "$path" ]]; do
+    cd "${path%/*}"
+    local name="${path##*/}"
+    path="$(bats_resolve_link "$name")"
+  done
+
+  printf -v "$2" -- '%s' "$PWD"
```

Comment:
There is a reason: `printf -v` will write to a variable, and in this case, to a variable in the caller's scope, the name of which the caller specifies as `$2`. To avoid name collisions with variables outside of the scope of functions like this, I try to use positional parameters directly (whereas I normally do assign them to local variables for clarity's sake).

But I'm just realizing I've got two other local variables here, and I'm not following my usual practice of mangling their names to reduce the likelihood of collision (where "mangling" here is prefixing variable names with something like `__bats_abs_path_` to reduce the likelihood of collision). I could also add such a local variable for `$2` as well.

Any preference?

Reply:
Thanks for the detail! Personal preference is to be explicit (I too forgot about `printf`'s variable assignment), so a "mangled reference" (sounds like that should be a programming term!) would be my suggestion.

Reply:
OK; I'll add those in #92. (Except for the one-liners, though.)

BTW, have you never done C++? "Name mangling" is a C++ implementation detail, but one that C++ coders need to be familiar with.

Reply:
Only casual commits, never anything serious. Glad to have learned it though 🤣

Reply:
So I pushed a new commit to #91, but only went so far, because the functions in `libexec/bats-exec-test` were mostly short, but all called frequently enough that there was a slight-yet-noticeable performance impact. I also decided not to mangle literally everything since these functions are all internal, so we control how they're called, and they're easier to read without all the mangling.